app/controllers/internal/metrics_controller.rb

```diff
@@ -9,16 +8,7 @@ class MetricsController < RestController::BaseController
       get '/internal/v4/metrics', :index
 
       def index
-        periodic_updater = VCAP::CloudController::Metrics::PeriodicUpdater.new(
-          Time.now.utc,
-          Steno::Sink::Counter.new,
-          Steno.logger('cc.api'),
-          [
-            VCAP::CloudController::Metrics::StatsdUpdater.new,
-            VCAP::CloudController::Metrics::PrometheusUpdater.new
-          ]
-        )
-        periodic_updater.update!
+        CloudController::DependencyLocator.instance.periodic_updater.update!
```

Comment:
Currently, whenever the metrics endpoint is being called, we will collect fresh metrics via PeriodicUpdater.update!. As for now there is only one scraper, which will call this endpoint, there won't be a performance problem with that. If in the future there might be several or even an unknown number of scrapers, which could access this endpoint, we might want to remove this update and serve the metrics, which have been periodically collected. TODO: If we update the metrics, each time the endpoint is being called, then the PeriodicUpdater does actually not need to collect metrics periodically for the Prometheus endpoint.

Reply:
I think I would simply remove this `update!` from here and return the values that are already periodically fetched.

Reply:
While working on visualising the new metrics I realised that we loose precision compared to the statsd stack due to the pull mechanism. I think we should leave the `update!` so people can choose the level of precision through changing the pull interval of the prom scraper.